runtime/drivers/postgres/sql_store.go

```diff
@@ -0,0 +1,190 @@
+package postgres
+
+import (
+	"context"
+	sqldriver "database/sql/driver"
+	"fmt"
+	"strings"
+
+	"github.com/jmoiron/sqlx"
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	"github.com/rilldata/rill/runtime/drivers"
+	"golang.org/x/exp/slices"
+
+	// load pgx driver
+	_ "github.com/jackc/pgx/v4/stdlib"
+)
+
+// Query implements drivers.SQLStore
+func (c *connection) Query(ctx context.Context, props map[string]any) (drivers.RowIterator, error) {
+	sql, ok := props["sql"].(string)
+	if !ok {
+		return nil, fmt.Errorf("property \"sql\" is mandatory for connector \"motherduck\"")
+	}
```

Comment:
1. Not motherduck
2. Consider adding `sourceProperties` struct like other connectors to make it easier to add new properties in the future